### PR TITLE
v2, podman: fix pod tests

### PR DESCRIPTION
**Pull request by giuseppe**
fix remaining failures with pod tests.

Only pause and pod stats are not addressed in this PR, since we are already tracking pause and pod stats in:

https://github.com/containers/libpod/pull/5975
https://github.com/containers/libpod/pull/5966

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: test/e2e/pod_infra_container_test.go [hunk 1]
```diff
@@ -95,7 +94,7 @@ var _ = Describe("Podman pod create", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session.ExitCode()).To(Equal(0))
 
-		check := podmanTest.Podman([]string{"ps", "-a", "--no-trunc", "--ns", "--format", "{{.IPC}} {{.NET}}"})
+		check := podmanTest.Podman([]string{"ps", "-a", "--no-trunc", "--ns", "--format", "{{.Namespaces.IPC}} {{.Namespaces.NET}}"})
```
Comment (vrothberg) [inline test/e2e/pod_infra_container_test.go:97]: The "Namespaces" changes seem like a breaking change. Is this known/intended? @mheon @baude

Comment (baude) [reply test/e2e/pod_infra_container_test.go:97]: the change is intended, though we should probably use methods to keep backwards compat?

Comment (vrothberg) [reply test/e2e/pod_infra_container_test.go:97]: +1 ... then we can make sure to test the new and old format here

Comment (giuseppe) [reply test/e2e/pod_infra_container_test.go:97]: good idea.  I've added methods to keep compatibility and fixed the tests to use both versions

File: libpod/container_internal_linux.go [hunk 2]
```diff
@@ -385,6 +385,16 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 			g.AddLinuxGIDMapping(uint32(0), uint32(0), uint32(1))
 		}
 	}
+
+	for _, i := range c.config.Spec.Linux.Namespaces {
+		if i.Type == spec.UTSNamespace {
```
Comment (mheon) [inline libpod/container_internal_linux.go:390]: Shouldn't we check for path here? To see if a path to the UTS namespace was given?

Comment (giuseppe) [reply libpod/container_internal_linux.go:390]: I've only moved the existing chunk.

When we join an existing container these values are overwritten in `addNamespaceContainer`

**Pull request merged**